--- a/libdwfl/Makefile.in
+++ b/libdwfl/Makefile.in
@@ -113,7 +113,7 @@
 	dwfl_getmodules.c dwfl_getdwarf.c dwfl_module_getdwarf.c \
 	dwfl_module_getelf.c dwfl_validate_address.c argp-std.c \
 	find-debuginfo.c dwfl_build_id_find_elf.c \
-	dwfl_build_id_find_debuginfo.c \
+	dwfl_build_id_find_debuginfo.c linux-kernel-modules.c \
 	linux-proc-maps.c dwfl_addrmodule.c dwfl_addrdwarf.c cu.c \
 	dwfl_module_nextcu.c dwfl_nextcu.c dwfl_cumodule.c \
 	dwfl_module_addrdie.c dwfl_addrdie.c lines.c dwfl_lineinfo.c \
@@ -142,7 +142,7 @@
 	dwfl_validate_address.$(OBJEXT) argp-std.$(OBJEXT) \
 	find-debuginfo.$(OBJEXT) dwfl_build_id_find_elf.$(OBJEXT) \
 	dwfl_build_id_find_debuginfo.$(OBJEXT) \
-	linux-proc-maps.$(OBJEXT) \
+	linux-kernel-modules.$(OBJEXT) linux-proc-maps.$(OBJEXT) \
 	dwfl_addrmodule.$(OBJEXT) dwfl_addrdwarf.$(OBJEXT) \
 	cu.$(OBJEXT) dwfl_module_nextcu.$(OBJEXT) \
 	dwfl_nextcu.$(OBJEXT) dwfl_cumodule.$(OBJEXT) \
@@ -402,7 +402,7 @@
 	dwfl_getmodules.c dwfl_getdwarf.c dwfl_module_getdwarf.c \
 	dwfl_module_getelf.c dwfl_validate_address.c argp-std.c \
 	find-debuginfo.c dwfl_build_id_find_elf.c \
-	dwfl_build_id_find_debuginfo.c \
+	dwfl_build_id_find_debuginfo.c linux-kernel-modules.c \
 	linux-proc-maps.c dwfl_addrmodule.c dwfl_addrdwarf.c cu.c \
 	dwfl_module_nextcu.c dwfl_nextcu.c dwfl_cumodule.c \
 	dwfl_module_addrdie.c dwfl_addrdie.c lines.c dwfl_lineinfo.c \
@@ -539,6 +539,7 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libdwfl_crc32_file.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/lines.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/link_map.Po@am__quote@
+@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/linux-kernel-modules.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/linux-core-attach.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/linux-pid-attach.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/linux-proc-maps.Po@am__quote@
