--- a/misc/util.c	2021-03-01 05:46:44.000000000 +0300
+++ b/misc/util.c	2021-03-01 12:08:40.941466300 +0300
@@ -105,7 +105,9 @@
 		printf(_("Proceed anyway (or wait %d seconds to proceed) ? (y,N) "),
 		       delay);
 		alarm(delay);
-	} else
+	}
+# if 0
+	  else
 		fputs(_("Proceed anyway? (y,N) "), stdout);
 	buf[0] = 0;
 	if (!fgets(buf, sizeof(buf), stdin) ||
@@ -115,6 +117,7 @@
 		putc('\n', stdout);
 		exit(1);
 	}
+#endif
 	signal(SIGALRM, SIG_IGN);
 }
 
