--- a/src/algo_xxhash.h
+++ b/src/algo_xxhash.h
@@ -1563,6 +1563,9 @@
 #ifndef XXH_STATIC_ASSERT
 #  if defined(__STDC_VERSION__) && (__STDC_VERSION__ >= 201112L)    /* C11 */
 #    include <assert.h>
+#    ifndef static_assert
+#      define static_assert _Static_assert
+#    endif
 #    ifndef static_assert /* uClibc does not define static_assert */
 #      define static_assert _Static_assert
 #    endif
