--- a/libxtables/xtables.c	2020-11-16 22:36:38.844548331 +0300
+++ b/libxtables/xtables.c	2020-11-16 22:37:06.216198145 +0300
@@ -403,6 +403,7 @@
 
 int xtables_insmod(const char *modname, const char *modprobe, bool quiet)
 {
+#if 0
 	char *buf = NULL;
 	char *argv[4];
 	int status;
@@ -437,6 +438,7 @@
 	free(buf);
 	if (WIFEXITED(status) && WEXITSTATUS(status) == 0)
 		return 0;
+#endif
 	return -1;
 }
 
